code/datums/components/pitfall.dm

```diff
@@ -0,0 +1,239 @@
+		RegisterSignal(src.parent, COMSIG_TURF_REPLACED, PROC_REF(RemoveComponent))
+		src.BruteDamageMax	= BruteDamageMax
+		src.HangTime		= HangTime
+
+	UnregisterFromParent()
+		. = ..()
+		UnregisterSignal(src.parent, COMSIG_ATOM_ENTERED)
+		UnregisterSignal(src.parent, COMSIG_TURF_LANDIN_THROWN)
+		UnregisterSignal(src.parent, COMSIG_ATTACKBY)
+		UnregisterSignal(src.parent, COMSIG_TURF_REPLACED)
+
+	/// returns the .parent but typecasted as a turf
+	proc/typecasted_parent()
+		RETURN_TYPE(/turf)
+		. = src.parent
+
+	/// updates targets for area/coordinate targeting. is overridden added to in child types.
+	proc/update_targets()
+		return
+
+	/// called when movable atom AM enters a pitfall turf. Mainly checks.
+	proc/start_fall(var/signalsender, var/atom/movable/AM)
+		if (!istype(AM, /atom/movable) || istype(AM, /obj/projectile))
+			return
+		if (AM.throwing) // throw em on over, why dont ya
+			return
+		if (AM.event_handler_flags & IMMUNE_PITFALL)
+			return
+		if (AM.flags & TECHNICAL_ATOM || istype(AM, /obj/blob)) //we can do this better (except the blob one, RIP)
+			return
+		if (AM.anchored >= ANCHORED_ALWAYS || (locate(/obj/lattice) in src.parent) || (locate(/obj/grille/catwalk) in src.parent))
+			return
+		if (ismob(AM))
+			var/mob/M = AM
+			if (M.client?.flying || isobserver(AM) || isintangible(AM) || istype(AM, /mob/wraith))
+				return
+
+		return_if_overlay_or_effect(AM)
+
+		// if the fall has coyote time, then delay it
+		if (src.HangTime)
+			SPAWN_DBG(src.HangTime)
+				if (!QDELETED(AM))
+					var/datum/component/pitfall/pitfall = AM.loc.GetComponent(/datum/component/pitfall)
+					pitfall?.try_fall(signalsender, AM)
+		else
+			src.try_fall(signalsender, AM)
+
+	/// called when it's time for movable atom AM to actually fall into the pit
+	proc/try_fall(var/signalsender, var/atom/movable/AM)
+		SHOULD_CALL_PARENT(TRUE)
+		return TRUE
+		// child procs will then use fall_to after calling this
+
+	/// a proc that makes a movable atom 'A' fall from 'src.typecasted_parent()' to 'T' with a maximum of 'brutedamage' brute damage
+	proc/fall_to(var/turf/T, var/atom/movable/AM, var/brutedamage = 50)
+		SHOULD_NOT_OVERRIDE(TRUE)
+		if(istype(AM, /obj/overlay) || AM.anchored == 2)
+			return
+		#ifdef CHECK_MORE_RUNTIMES
+		if(current_state <= GAME_STATE_WORLD_NEW)
+			CRASH("[identify_object(AM)] fell into [src.typecasted_parent()] at [src.typecasted_parent().x],[src.typecasted_parent().y],[src.typecasted_parent().z] ([src.typecasted_parent().loc] [src.typecasted_parent().loc.type]) during world initialization")
+		#endif
+		src.typecasted_parent().visible_message(SPAN_ALERT("[AM] falls into [src.typecasted_parent()]!"))
+		if(src.FallTime)
+			animate_fall(AM,src.FallTime,src.DepthScale)
+			var/old_anchored = AM.anchored
+			var/old_density = AM.density
+			var/old_canmove
+			var/mob/M
+			if(ismob(AM))
+				M = AM
+				old_canmove = M.canmove
+				M.canmove = 0
```

Comment:
Didn't find this to be working, and I could stack 6 falls in the time it took me to go down a crag elevator (which, also funny)

It seems the usual way of doing this is with a mob property PROP_CANTMOVE. Makes me wonder if other stuff directly setting canmove is broken too.

Reply:
Did a bit of testing, directly setting canmove doesn't work, mhm. The canmove life process resets it.